### PR TITLE
resolve bug in check

### PR DESCRIPTION
**Pull request by owades**
# Description
By joining on both the vp and tu keys, I was filtering out non-null values in each. This was fine for the tu feed, since the check only cares about trips that appear in the tu feed. I removed the join to the vp feed, and instead joined on the associated schedule feed - I believe this should yield the same result.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
Confirmed that some checks both pass and fail.
Ran ` ./dbt.sh build -s int_gtfs_quality__all_tu_in_vp+ `

### CODE REVIEW

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__all_tu_in_vp.sql [hunk 1]
```diff
@@ -39,7 +39,9 @@ joined AS (
     LEFT JOIN fct_observed_trips AS f
     ON idx.date = f.dt
     AND quartet.trip_updates_gtfs_dataset_key = f.tu_gtfs_dataset_key
-    AND quartet.vehicle_positions_gtfs_dataset_key = f.vp_gtfs_dataset_key
+    -- Joining on vp_gtfs_dataset_key would ensure that vp_num_distinct_message_ids was never null
+    -- The below join condition may be unnecessary, though it could rule out potential cases where one TU feed is mapped to multiple scheduel feeds
```
Comment (lauriemerrell) [inline warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__all_tu_in_vp.sql:43]: Massive nit: typo in `scheduel`

**Review by lauriemerrell — COMMENTED** [review]
Just to confirm: Is the intention that if there's a trip updates feed with no vehicle positions feed at all, that should fail? And if there's a vehicle positions feed with no trip updates, is the result supposed to be `N/A`? I wonder if it's worth just double checking each of those cases to ensure it's coming out as expected.

Here's a query identifying this (you can swap tu and vp to check the other one):

```
SELECT *
FROM `cal-itp-data-infra.mart_transit_database.dim_provider_gtfs_data`
WHERE vehicle_positions_gtfs_dataset_key IS NULL AND trip_updates_gtfs_dataset_key IS NOT NULL AND _is_current
ORDER BY organization_name
```

Apologies for misleading, I guess I had assumed that the intention was "if both trip updates and vehicle positions are present, this condition should be met" (which, annoying as it is, is what the indices help clarify: what is our universe for each check... because this is implemented as a `service` check, we can kind of do any combination)

**Comment by owades** [comment]
> Just to confirm: Is the intention that if there's a trip updates feed with no vehicle positions feed at all, that should fail? And if there's a vehicle positions feed with no trip updates, is the result supposed to be `N/A`? I wonder if it's worth just double checking each of those cases to ensure it's coming out as expected.
> 
> Here's a query identifying this (you can swap tu and vp to check the other one):
> 
> ```
> SELECT *
> FROM `cal-itp-data-infra.mart_transit_database.dim_provider_gtfs_data`
> WHERE vehicle_positions_gtfs_dataset_key IS NULL AND trip_updates_gtfs_dataset_key IS NOT NULL AND _is_current
> ORDER BY organization_name
> ```
> 
> Apologies for misleading, I guess I had assumed that the intention was "if both trip updates and vehicle positions are present, this condition should be met" (which, annoying as it is, is what the indices help clarify: what is our universe for each check... because this is implemented as a `service` check, we can kind of do any combination)

Thank you for looking through this and thinking critically about it as usual. I think the original design may have been correct, and that we might need updates to `fct_observed_trips`. In hindsight this PR is basically a hack to get directionally accurate results.

The issue is that as it stands, if there are no vehicle positions for a given trip for a given day, `fct_observed_trips` will have a null value for the associated `vp_gtfs_dataset_key`. I think we need to switch the order of the joins in the `vehicle_positions_with_associated_schedule` [subquery](https://github.com/cal-itp/data-infra/blob/f97681a8d808cce6a58a6f5cb3ca2220219926ba/warehouse/models/mart/gtfs/fct_observed_trips.sql#L69). @atvaccaro

**Review by lauriemerrell — APPROVED** [review]

**Comment by lauriemerrell** [comment]
Talked to @owades offline, we are just going to merge this and revisit again later

**Pull request merged**